dandi/organize.py

```diff
@@ -172,6 +173,81 @@ def create_unique_filenames_from_metadata(metadata):
     return metadata
 
 
+def _create_external_file_names(metadata: dict):
+    """
+    Renames the external_file attribute in an ImageSeries according to the rule:
+    Example, the Initial name of file:
+        external_file = [name1.mp4]
+    rename to:
+        external_file = [dandiset-path-of-nwbfile/
+                dandi-renamed-nwbfile_name(folder without extension .nwb)/
+                f'{ImageSeries.object_id}_external_file_0.mp4'
+    This is stored in a new field in the
+    metadata['external_file_objects'][0]['external_files_renamed']
+    Parameters
+    ----------
+    metadata: list
+        list of metadata dictionaries created during the call to pynwb_utils._get_pynwb_metadata
+    Returns
+    -------
+    metadata: list
+        updated list of metadata dictionaries
+    """
+    metadata = deepcopy(metadata)
+    for meta in metadata:
+        if "dandi_path" not in meta or "external_file_objects" not in meta:
+            continue
+        nwb_folder_name = op.basename(meta["dandi_path"]).split(".")[0]
```

Comment:
What if the basename begins with a period?

Reply:
the basename is the name of an nwbfile as renamed by dandi so its not expected to start with a period.

Reply:
not expected indeed, but i would not rely on that 100%.
```suggestion
        nwb_folder_name = op.splitext(op.basename(meta["dandi_path"]))[0]
```
?